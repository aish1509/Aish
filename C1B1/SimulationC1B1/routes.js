export const URLS = {
	ROOT: '/albumpage',
	VIDEO: '/video',
	MUSICGENRE: '/musicgenre',
	RAISEFUNDS: '/raisefunds',
	SETUPREWARDS: '/setrewards',
	SHAREVIDEO: '/sharevideo',
	CHOSETIME: '/chosetime',
	RESULT: '/result'
};
