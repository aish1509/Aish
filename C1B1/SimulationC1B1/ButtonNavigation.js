import React from 'react';
import PropTypes from 'prop-types';
import {URLS} from './routes';
import {MathButton} from '@math/common';

const ButtonNavigation = ({path, history}) => {
	const onClickHandlerGoOn = () => {
		if (history.location.pathname === `${path}${URLS.VIDEO}`) {
			history.push(`${path}${URLS.MUSICGENRE}`);
			return;
		}
		if (history.location.pathname === `${path}${URLS.MUSICGENRE}`) {
			history.push(`${path}${URLS.RAISEFUNDS}`);
			return;
		}
		if (history.location.pathname === `${path}${URLS.RAISEFUNDS}`) {
			history.push(`${path}${URLS.SETUPREWARDS}`);
			return;
		}
		if (history.location.pathname === `${path}${URLS.SETUPREWARDS}`) {
			history.push(`${path}${URLS.SHAREVIDEO}`);
			return;
		}
		if (history.location.pathname === `${path}${URLS.SHAREVIDEO}`) {
			history.push(`${path}${URLS.CHOSETIME}`);
			return;
		}
		if (history.location.pathname === `${path}${URLS.CHOSETIME}`) {
			history.push(`${path}${URLS.RESULT}`);
		}
	};

	const onClickHandlerStart = () => {
		history.push(`${path}${URLS.VIDEO}`);
	};

	const onClickHandlerStartOver = () => {
		history.push(`${path}${URLS.ROOT}`);
	};

	const onClickHandlerExit = () => {
		history.push('/explorezone');
	};

	const styles = {
		float: 'right',
		marginRight: '2%'
	};
	if (history.location.pathname !== `${path}${URLS.ROOT}`) {
		return (
			<>
				{history.location.pathname !== `${path}${URLS.RESULT}` && (
					<MathButton
						data-testid="Button-Go-On"
						buttonStyleModifier={styles}
						onClickHandler={onClickHandlerGoOn}
					>
						{history.location.pathname ===
						`${path}${URLS.CHOSETIME}`
							? 'Run'
							: 'Go On'}
					</MathButton>
				)}
				{history.location.pathname === `${path}${URLS.RESULT}` && (
					<MathButton
						data-testid="Button-Exit"
						buttonStyleModifier={styles}
						onClickHandler={onClickHandlerExit}
					>
						Exit
					</MathButton>
				)}
				<MathButton
					data-testid="Button-StartOver"
					buttonStyleModifier={styles}
					onClickHandler={onClickHandlerStartOver}
					secondary={true}
				>
					Start Over
				</MathButton>
			</>
		);
	}
	return (
		<MathButton
			data-testid="Button-Start"
			buttonStyleModifier={styles}
			onClickHandler={onClickHandlerStart}
		>
			Start
		</MathButton>
	);
};

ButtonNavigation.propTypes = {
	path: PropTypes.string.isRequired,
	history: PropTypes.object.isRequired
};

export default ButtonNavigation;
