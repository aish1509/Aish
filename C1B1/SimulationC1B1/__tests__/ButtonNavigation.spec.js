import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {URLS} from '../routes';
import ButtonNavigation from '../index';

describe('<ButtonNavigation />', () => {
	let wrapper = null;
	let mockPath = null;
	let mockHistory = null;
	let mockOnClickHandlerGoOn = null;
	let mockOnClickHandlerStart = null;
	let mockOnClickHandlerStartOver = null;
	let mockOnClickHandlerExit = null;

	beforeEach(() => {
		mockPath = '/explorezone/simulation';
		mockHistory = {
			location: {pathname: '/explorezone/simulation/albumpage'}
		};
		mockOnClickHandlerGoOn = jest.fn();
		mockOnClickHandlerStart = jest.fn();
		mockOnClickHandlerStartOver = jest.fn();
		mockOnClickHandlerExit = jest.fn();
	});

	describe('function props are passed in', () => {
		wrapper = shallow(
			<ButtonNavigation path={mockPath} history={mockHistory} />
		);
	});

	it('should render correctly', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('should render video page', () => {
		mockHistory = {
			location: {pathname: `/explorezone/simulation/${URLS.VIDEO}`}
		};
		wrapper.setProps({history: mockHistory});
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
	const buttonComp = wrapper.find('MathButton');
	expect(buttonComp).toHaveLength(1);
	//const mathbutton = wrapper.find('MathButton');
	it('onClickHandler fires correctly', () => {
		buttonComp
			.find('[data-testid="Button-Go-On"]')
			.prop('onClickHandler')();
		expect(mockOnClickHandlerGoOn).toHaveBeenCalled();
	});
	it('onClickHandler fires correctly', () => {
		buttonComp.find('[data-testid="Button-Exit"]').prop('onClickHandler')();
		expect(mockOnClickHandlerExit).toHaveBeenCalled();
	});
	it('onClickHandler fires correctly', () => {
		buttonComp
			.find('[data-testid="Button-StartOver"]')
			.prop('onClickHandler')();
		expect(mockOnClickHandlerStartOver).toHaveBeenCalled();
	});
	it('onClickHandler fires correctly', () => {
		buttonComp
			.find('[data-testid="Button-Start"]')
			.prop('onClickHandler')();
		expect(mockOnClickHandlerStart).toHaveBeenCalled();
	});
});
