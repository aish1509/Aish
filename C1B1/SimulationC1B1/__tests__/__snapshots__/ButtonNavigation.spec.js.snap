// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonNavigation /> should render correctly 1`] = `null`;
